### ISSUE TITLE
[Bug]: System settings for version history are ignored in the VersionCleanupTask when stored in the settings-store

### ISSUE THREAD

**Issue by fuw-nielsjensen** [pimcore/pimcore#16877]
### Pimcore version

v11.2.2

### Steps to reproduce

1. Install the pimcore demo application
2. Make several changes on a document to create some versions
3. Set 'Store version history for x Steps:' in the system settings to a lower value than the number of versions you have created 
4. Run `bin/console pimcore:maintenance -j versioncleanup` and `bin/console messenger:consume pimcore_core pimcore_maintenance`

=> The versions **are not deleted** from the database

5. Switch to 'symfony-config' for the location of system_settings:
```yaml
# /config/config.yaml

pimcore:
    config_location:
        system_settings:
            write_target:
                type: 'symfony-config'
            read_target:
                type: 'symfony-config'
```
6. Save 'Store version history for x Steps:' in the admin interface again (settings are saved in `/var/config/system_settings/system_settings.yaml`)
7. Run `bin/console pimcore:maintenance -j versioncleanup` and `bin/console messenger:consume pimcore_core pimcore_maintenance` again

=> The versions **are deleted** from the database this time

### Actual Behavior

The versions are not deleted when the configuration is saved in the settings-store.

### Expected Behavior

The VersionCleanupTask should respect the settings, when saved in the settings-store.

Replacing the injected \Pimcore\Config class in the [VersionCleanupTask](https://github.com/pimcore/pimcore/blob/ef68e4543bb82cf3b9af1117a91f00379978fc3f/lib/Maintenance/Tasks/VersionsCleanupTask.php) with \Pimcore\SystemSettingsConfig works for me:

```php
    // ...
    private SystemSettingsConfig $config;

    // ...

    public function __construct(LoggerInterface $logger, SystemSettingsConfig $config)
    {
        $this->logger = $logger;
        $this->config = $config;
    }

    // ...

    private function doVersionCleanup(): void
    {
        $systemSetttingsConfig = $this->config->getSystemSettingsConfig();
        $conf = [
            'document' => $systemSetttingsConfig['documents']['versions'] ?? null,
            'asset' => $systemSetttingsConfig['assets']['versions'] ?? null,
            'object' => $systemSetttingsConfig['objects']['versions'] ?? null
        ];
        // ...
    }
```

**Comment by blankse** [comment]
@fuw-nielsjensen I created a PR: https://github.com/pimcore/pimcore/pull/16883

**Comment by blankse** [comment]
Fixed by #16883
